assets/js/components/notifications/nodata.js

```diff
@@ -41,6 +43,28 @@ const getNoDataComponent = ( moduleName, inGrid = false, fullWidth = false, crea
 		description={ sprintf( __( '%s data is not yet available, please check back later.', 'google-site-kit' ), moduleName ) }
 	/>;
 
+	if ( identifier && googlesitekit.modules[ identifier ] ) {
+		const {
+			active,
+			setupComplete
+		} = googlesitekit.modules[ identifier ];
+
+		if ( active && ! setupComplete ) {
+			cta = <CTA
+
+				/* translators: %s: Module name */
+				title={ sprintf( __( '%s activation', 'google-site-kit' ), moduleName ) }
+
+				/* translators: %s: Module name */
+				description={ sprintf( __( '%s module needs to be configured.', 'google-site-kit' ), moduleName ) }
+				onClick={ () => {
+					window.location = getSiteKitAdminURL( `googlesitekit-module-${identifier}` );
+				} }
+				ctaLabel={ __( 'Complete activation', 'google-site-kit' ) }
+			/>;
+		}
+	}
+
```

Comment:
Instead of making the `getNoDataComponent()` function more complex, let's create a new function dedicated to it. Generally, a single parameter shouldn't decide on the entire function to do something different.

I'd suggest creating a `getSetupIncompleteComponent()` function and calling it in the respective areas. Especially since we check these `.active` and `! .setupComplete` conditions in these areas anyway, we can simply call the new function from there.

Reply:
thanks @felixarntz I have pushed an update with getSetupIncompleteComponent()